app/controllers/api/v1/user_favourites_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V1
+    class UserFavouritesController < Api::V1::ApiController
+      load_and_authorize_resource :user_favourite, parent: false
+
+      resource_description do
+        formats ['json']
+        error 401, "Unauthorized"
+        error 404, "Not Found"
+        error 422, "Validation Error"
+        error 500, "Internal Server Error"
+      end
+
+      api :GET, '/v1/user_favourites', "Fetches user_favourites"
+      def index
+        user_favourites = @user_favourites.where(user_id: current_user.id)
```

Comment:
You can let cancan handle the user filtering part (look at comment I wrote in ability.rb)

Reply:
sure.. ill add this change.